netbox/ipam/models/ip.py

```diff
@@ -866,9 +866,16 @@ def clean(self):
 
         # Check for primary IP assignment that doesn't match the assigned device/VM
         if self.pk:
-            for cls, attr in ((Device, 'device'), (VirtualMachine, 'virtual_machine')):
+            for cls, attr in ((Device, 'device'), (VirtualMachine, 'virtual_machine'), (VirtualDeviceContext, 'vdcs')):
```

Comment:
This might be a bit cleaner - I think line 872 comparison of .all() != parent might be wrong, should be first()?  I'm assuming the m2m check only needs to be done if 'vdcs', device and virtual_machine attrs aren't m2m.
```
        if self.pk:
            for cls, attr in ((Device, 'device'), (VirtualMachine, 'virtual_machine'), (VirtualDeviceContext, 'vdcs')):
                parent = cls.objects.filter(Q(primary_ip4=self) | Q(primary_ip6=self)).first()
                if parent and hasattr(self.assigned_object, attr):
                    if attr == 'vdcs'and getattr(self.assigned_object, attr).first() != parent:
                        raise ValidationError({
                            'interface': f"IP address is primary for {cls._meta.model_name} {parent} but "
                                         f"not assigned to it!"
                        })
                    elif getattr(self.assigned_object, attr, None) != parent:
                        # Check for a NAT relationship
                        if not self.nat_inside or getattr(self.nat_inside.assigned_object, attr, None) != parent:
                            raise ValidationError({
                                'interface': f"IP address is primary for {cls._meta.model_name} {parent} but "
                                             f"not assigned to it!"
                            })

```

Reply:
also, move the parent = cls.objects outside the loop:
```
        if self.pk:
            parent = cls.objects.filter(Q(primary_ip4=self) | Q(primary_ip6=self)).first()
            for cls, attr in ((Device, 'device'), (VirtualMachine, 'virtual_machine'), (VirtualDeviceContext, 'vdcs')):

```